cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java

```diff
@@ -142,4 +165,132 @@ public void setMissingIncludeWarningsEnabled(boolean enabled){
+        while ((line = br.readLine()) != null) {
+          if (line.startsWith("  INCLUDE=")) { // handle environment includes 
+            String[] includes = line.split("=")[1].split(";");
+            for(String include : includes) {
+              if (!uniqueIncludes.contains(include)) {
+                uniqueIncludes.add(include);
+              }
+            }
+          }
+          
+          // get base path of project to make 
+          // Target "ClCompile" in file "C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppCommon.targets" from project "D:\Development\SonarQube\cxx\sonar-cxx\integration-tests\testdata\googletest_bullseye_vs_project\PathHandling.Test\PathHandling.Test.vcxproj" (target "_ClCompile" depends on it):
+          if (line.startsWith("Target \"ClCompile\" in file")) {
+            currentProjectPath = line.split("\" from project \"")[1].split("\\s+")[0].replace("\"", "");              
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\BullseyeCoverage\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);
+          }
+          
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 10.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 11.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 11.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);       
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 12.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);        
+          }
+          if (line.contains("C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe") || 
+                  line.contains("C:\\Program Files\\Microsoft Visual Studio 14.0\\VC\\bin\\CL.exe")) {
+            parseVCppCompilerCLLine(line, currentProjectPath);        
+          }  
+        }
+        br.close();
+      } catch (IOException ex) {
+        LOG.error("Cannot parse build log", ex);
+      }
+  }
+
+  private void parseVCppCompilerCLLine(String line, String projectPath) {
+    File file = new File(projectPath);
+    String project = file.getParent();
+
+    for (String includeElem : getMatches(Pattern.compile("/I\"(.*?)\""), line)) {
+      ParseInclude(includeElem, project);
+    }
+
+    for (String includeElem : getMatches(Pattern.compile("/I([^\\s\"]+) "),
+        line)) {
+      ParseInclude(includeElem, project);
+    }
+
+    for (String macroElem : getMatches(Pattern.compile("[/-]D\\s([^\\s]+)"),
```

Comment:
Ok good I try to include this, later I need to test this in our 2013 build.
But this should be a good start definitely

On Fri, Apr 10, 2015, 14:50 Günter Wirth <notifications@github.com> wrote:

> In cxx-squid/src/main/java/org/sonar/cxx/CxxConfiguration.java
> <https://github.com/wenns/sonar-cxx/pull/399#discussion_r28138691>:
>
> > +  }
> > +
> > +  private void parseVCppCompilerCLLine(String line, String projectPath) {
> > +    File file = new File(projectPath);
> > +    String project = file.getParent();
> > +
> > +    for (String includeElem : getMatches(Pattern.compile("/I\"(.*?)\""), line)) {
> > +      ParseInclude(includeElem, project);
> > +    }
> > +
> > +    for (String includeElem : getMatches(Pattern.compile("/I([^\\s\"]+) "),
> > +        line)) {
> > +      ParseInclude(includeElem, project);
> > +    }
> > +
> > +    for (String macroElem : getMatches(Pattern.compile("[/-]D\\s([^\\s]+)"),
>
> Below an Python example how to do this. That's maybe a good starting point:
>
> # VS 2010 predefined macros:# http://msdn.microsoft.com/en-us/library/vstudio/b0084kay(v=vs.100).aspx
> def macro_expansion(line,logobj,target):
>     vs_macros = {}
>
>     if target=='64':
>         vs_macros.update({'__ptr32=':''})
>
>     if '/GR-' not in line:
>         vs_macros.update({'_CPPRTTI':''})
>
>     if '/EHs ' in line:
>         vs_macros.update({'_CPPUNWIND':''})
>     elif '/EHa ' in line:
>         vs_macros.update({'_CPPUNWIND':''})
>     elif '/EHsc ' in line:
>         vs_macros.update({'_CPPUNWIND':''})
>     elif '/EHac ' in line:
>         vs_macros.update({'_CPPUNWIND':''})
>
>     if '/MD ' in line:
>         vs_macros.update({'_MT':''})
>         vs_macros.update({'_DLL':''})
>
>     if '/MDd' in line:
>         vs_macros.update({'_DEBUG':''})
>         vs_macros.update({'_MT':''})
>         vs_macros.update({'_DLL':''})
>
>     if '/MT ' in line:
>         vs_macros.update({'_MT':'0'})
>
>     if '/MTd' in line:
>         vs_macros.update({'_DEBUG':''})
>         vs_macros.update({'_MT':''})
>
>     if '/LD ' in line:
>         vs_macros.update({'_MT':''})
>
>     if '/LDd' in line:
>         vs_macros.update({'_MT':''})
>         vs_macros.update({'_DEBUG':''})
>
>     if '/clr' in line:
>         vs_macros.update({'_M_CEE':''})
>         vs_macros.update({'_MANAGED':''})
>
>     if '/clr:pure' in line:
>         vs_macros.update({'_M_CEE_PURE':''})
>
>     if '/clr:safe' in line:
>         vs_macros.update({'_M_CEE_SAFE':''})
>
>     if target=='32':
>         if '/arch:IA32' in line:
>             vs_macros.update({'_M_IX86_FP':'0'})
>         elif '/arch:SSE' in line:
>             vs_macros.update({'_M_IX86_FP':'1'})
>         elif '/arch:SSE2' in line:
>             vs_macros.update({'_M_IX86_FP':'2'})
>         elif '/arch:AVX' in line:
>             vs_macros.update({'_M_IX86_FP':'2'})
>             vs_macros.update({'__AVX__':''})
>         elif '/arch:AVX2' in line:
>             vs_macros.update({'_M_IX86_FP':'2'})
>             vs_macros.update({'__AVX__':''})
>             vs_macros.update({'__AVX2__':''})
>         else:
>             vs_macros.update({'_M_IX86_FP':'0'})
>
>     if '/Ze' not in line:
>         vs_macros.update({'_MSC_EXTENSIONS':'1'})
>
>     if '/RTC' in line or '/GZ' in line:
>         vs_macros.update({'__MSVC_RUNTIME_CHECKS':''})
>
>     if '/Zc:wchar_t' in line:
>         vs_macros.update({'_NATIVE_WCHAR_T_DEFINED':''})
>         vs_macros.update({'_WCHAR_T_DEFINED':''})
>
>     if '/Wp64' in line:
>         vs_macros.update({'_Wp64':''})
>
>     if '/Zl' in line:
>         vs_macros.update({'_VC_NODEFAULTLIB':''})
>
>     if '/G5' in line:
>         vs_macros.update({'_M_IX86':'500'})
>     elif '/G6' in line or '/GB' in line:
>         vs_macros.update({'_M_IX86':'600'})
>     elif '/G3' in line:
>         vs_macros.update({'_M_IX86':'300'})
>     elif '/G4' in line:
>         vs_macros.update({'_M_IX86':'400'})
>
>     return vs_macros
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wenns/sonar-cxx/pull/399/files#r28138691>.
>